### PR TITLE
Tests for creating simulated data and handling edge cases in running multiple CM analysis

### PR DESCRIPTION
**Pull request by azimov**
Collection of unit test additions including:
* Running multiple analyses with separable interaction terms and non-seperable interaction terms (computed by making entire population a single gender and defining a analysis to be based on gender)
* Checks on running multiple analysis with bad inputs
* Creating simulation profile
* Error/checks for creating simulation profile without covariates or cohorts
* Warning for limited cohort data that may not be useful in a simulation

Some warnings are still coming from upstream packages (mainly Andromeda/Cyclops) so warnings are checked with capture warnings.

### CODE REVIEW

File: tests/testthat/test-simulation.R [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                                            washoutPeriod = 183,
+                                            firstExposureOnly = TRUE,
+                                            removeDuplicateSubjects = TRUE,
+                                            restrictToCommonPeriod = TRUE,
+                                            maxCohortSize = 100000,
+                                            covariateSettings = covarSettings)
+
+
+  cohorts <- data.frame(cohortMethodData$cohorts)
+  cohorts$daysToCohortEnd <- rexp(nrow(cohorts), rate = 10)
```
Comment (msuchard) [inline tests/testthat/test-simulation.R:40]: Very inspired!

**Review by msuchard — APPROVED** [review]

**Review by msuchard — CHANGES_REQUESTED** [review]
I see this in R-check:
```
  createCohortMethodDataSimulationProfile: no visible binding for global
291
    variable 'daysToCohortEnd'
292
  Undefined global functions or variables:
293
    daysToCohortEnd
```   
 suggesting that you should use `.data$daysToCohortEnd` somewhere.

**Review by msuchard — CHANGES_REQUESTED** [review]
R-check also shows:


```
── Error (test-simulation.R:43:3): createCohortMethodDataSimulationProfile ─────
2086
Error: Error: no applicable method for 'count' applied to an object of class "NULL"
2087
Backtrace:
2088
    █
2089
 1. ├─CohortMethod::createCohortMethodDataSimulationProfile(cohortMethodData) test-simulation.R:43:2
2090
 2. │ └─cohortMethodData$covarites %>% count() %>% pull()
2091
 3. ├─dplyr::pull(.)
2092
 4. └─dplyr::count(.)
2093
── Error (test-simulation.R:71:3): Test bad covariate data ─────────────────────
```
suggesting that `covarites` is a misspelling somewhere.

**Review by msuchard — APPROVED** [review]

**Comment by azimov** [comment]
There remains one failing test on the OSX build:
```
  ══ Failed tests ════════════════════════════════════════════════════════════════
  ── Failure (test-parameterSweep.R:221:3): Outcome functions ────────────────────
  length(unique(logRrs)) not equal to length(logRrs).
  1/1 mismatches
  [1] 12 - 13 == -1
```

This is a strange failure as it doesn't happen on my local Mac. There are also no changes to the code in this branch that would impact this, as far as I can tell.

**Comment by msuchard** [comment]
@azimov  -- i am going to merge this `pr` but please bare in mind that we are getting a mac-os / GitHub error.

**Comment by azimov** [comment]
> @azimov -- i am going to merge this `pr` but please bare in mind that we are getting a mac-os / GitHub error.

@msuchard  yep, I'm looking into why this might be. There are some warnings that are thrown leading up to this on both my local mac and the GitHub actions script (integer overflow errors).

```
── Warning (test-parameterSweep.R:193:9): Outcome functions ────────────────────
NAs produced by integer overflow
Backtrace:
 1. CohortMethod::fitOutcomeModel(...) test-parameterSweep.R:193:8
 7. Cyclops:::confint.cyclopsFit(fit, parm = treatmentVarId, includePenalty = TRUE)
 9. bit64::as.integer.integer64(parm)
─
```

**Comment by msuchard** [comment]
@azimov -- i can reproduce this warnings on my local mac via `devtools::test(filter = "parameterSweep")`

**Comment by azimov** [comment]
> @azimov -- i can reproduce this warnings on my local mac via `devtools::test(filter = "parameterSweep")`

So I'm not sure how bit64 is supposed to function but the following seems strange to me. Passing 0 is fine but passing any other number results in NA


```
> bit64::as.integer.integer64(0)
[1] 0
> bit64::as.integer.integer64(1)
Warning: NAs produced by integer overflow
[1] NA
> bit64::as.integer.integer64(10000)
Warning: NAs produced by integer overflow
[1] NA
> bit64::as.integer.integer64(10000.0)
Warning: NAs produced by integer overflow
[1] NA
> bit64::as.integer.integer64(0.0)
[1] 0
> bit64::as.integer.integer64(0.0001)
Warning: NAs produced by integer overflow
[1] NA
> bit64::as.integer.integer64(-1)
Warning: NAs produced by integer overflow
[1] NA
> bit64::as.integer.integer64(-65)
Warning: NAs produced by integer overflow
[1] NA

```
Should cyclops be using `bit64::as.integer64 instead of bit64::as.initeger.integer64`?

**Comment by schuemie** [comment]
This confusion unfortunately stems from R allow periods in function names and defining generics using periods.

The generic is `as.integer()`, which can be applied to many types of arguments. The `as.integer.integer64()` function implements the function specifically for arguments of type `integer64`. It shouldn't really be called directly, rather be automatically invoked when calling `as.integer()` on an `integer64` object.

**Comment by schuemie** [comment]
I guess you're looking for `bit64::as.integer64`?
